app/services/provider_enrichments/rollover_enrichment_to_provider_service.rb

```diff
@@ -0,0 +1,11 @@
+module ProviderEnrichments
+  class RolloverEnrichmentToProviderService
+    def execute(enrichment:, new_provider:)
+      new_enrichment = enrichment.dup
+      new_enrichment.last_published_at = nil
+      new_enrichment.provider = new_provider
+      new_enrichment.save!
+      new_enrichment.rolled_over!
```

Comment:
I believe `rolled_over!` calls `update!` which calls `save!`. Is it worth changing this to:

```ruby
      new_enrichment = enrichment.dup
      new_enrichment.last_published_at = nil
      new_enrichment.provider = new_provider
      new_enrichment.rolled_over!
```

Just on principle of hitting the DB less?